src/websocket/server.cc

```diff
@@ -140,6 +141,21 @@ future<> connection::read_http_upgrade_request() {
         if (upgrade_header != "websocket") {
             throw websocket::exception("Upgrade header missing");
         }
+
+        sstring subprotocol = req->get_header("WebSocket-Subrotocol");
+        if (subprotocol.empty()) {
+            // TODO: disabled temporarily for testing
+            //throw websocket::exception("Subprotocol header missing.");
+            subprotocol = "echo";
```

Comment:
I suppose it's just a leftover now? It looks like you already set a proper subprotocol in the unit test, so this hack can be removed